ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -210,20 +110,20 @@ def _initialize_trainers(self, trainer_config):
                 for k in trainer_config[_brain_key]:
                     trainer_parameters[k] = trainer_config[_brain_key][k]
             trainer_parameters_dict[brain_name] = trainer_parameters.copy()
-        for brain_name in self.env.external_brain_names:
+        for brain_name in self.external_brains:
             if trainer_parameters_dict[brain_name]['trainer'] == 'offline_bc':
                 self.trainers[brain_name] = OfflineBCTrainer(
-                    self.env.brains[brain_name],
```

Comment:
Make sure BC works, I think there might be a bug here. `self.env.brains` contains all the brains, not just the external ones.

Reply:
This is getting the `brain_name` from `self.external_brains` and just looking them up them in `self.env.brains`, so I think it should be fine.

I've double checked and it works as expected for PushBlockIL